lib/mixlib/shellout/windows.rb

```diff
@@ -321,31 +321,43 @@ def self.executable?(path)
           File.executable?(path) && !File.directory?(path)
         end
 
+        def self.system_required_processes
+          [
+            'System Idle Process',
+            'System',
+            'spoolsv.exe',
+            'lsass.exe',
+            'csrss.exe',
+            'smss.exe',
+            'svchost.exe'
+          ]
+        end
+
         # recursively kills all child processes of given pid
         # calls itself querying for children child procs until
         # none remain. Important that a single WmiLite instance
         # is passed in since each creates its own WMI rpc process
         def self.kill_process_tree(pid, wmi, logger)
           wmi.query("select * from Win32_Process where ParentProcessID=#{pid}").each do |instance|
+            next if system_required_processes.include? instance.wmi_ole_object.name
```

Comment:
Does this happen often? If not always-ish, can we have a debug log here that we wanted to kill something we shouldn't so maybe in the future we can find a better solution than a whitelist?